### ISSUE TITLE
flake8 throws FutureWarning

### ISSUE THREAD

**Issue by dmohns** [Homebrew/homebrew-core#33370]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
- What happened (include command output)
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew install` commands)

## What you were trying to do (and why)
Install `flake8` via `brew` and running it throws a `FutureWarning`.

## Why is it a problem?
This Warning prevents the flake8 installation to be used in `atom-linter` plugin.

## Steps to re-produce

```sh
brew update
brew install flake8
flake
```
Output
```
/usr/local/Cellar/flake8/3.5.0_1/libexec/lib/python3.7/site-packages/pycodestyle.py:113: FutureWarning: Possible nested set at position 1
  EXTRANEOUS_WHITESPACE_REGEX = re.compile(r'[[({] | []}),;:]')
```

**Comment by dmohns** [comment]
### First analysis of the root cause:
`flake8` depends on `pycodestyle`. `pycodestyle` only has Python 3 support after version `>=2.4.0`. See https://github.com/PyCQA/pycodestyle/issues/728 and https://github.com/PyCQA/pycodestyle/pull/735.
The `flake8` version `3.5.0` that we currently install depends on `pycodestle<2.4.0` and has hence no full Python3 support. See https://github.com/PyCQA/flake8/blob/3.5.0/setup.py#L25
This commit 2df2844bb758eff74f55c939f89c6ccdffeb8319 migrated flake8 to Python 3. 

### Possible solutions

1. Rollback Python 3 migration and run flake8 using Python 2
2. Upade `flake8` to version `>=3.6.0` which should have full Python 3 support

**Comment by fxcoudert** [comment]
Fixed by #33371
Thanks for the bug report and the pull request, you rock!
